packages/react-server/src/ReactFizzHooks.js

```diff
@@ -386,13 +386,11 @@ export function useReducer<S, I, A>(
         } while (update !== null);
 
         // $FlowFixMe[incompatible-use] found when upgrading Flow
-        workInProgressHook.memoizedState = newState;
-
-        return [newState, dispatch];
+        workInProgressHook.memoizedState = [newState, dispatch];
```

Comment:
now naming for memoizedState is confusing - it is not only state but also dispatch

Reply:
You might be confused because there's the word "state" in both places, but "memoizedState" refers to the hook's internal state, not the state in the "useState's return value" sense. In `useMemo`, `.memoizedState` stores the hook state as `[value, previousDeps]` even though only `value` is returned from the hook. In `useRef`, it stores the ref object.